src/pubsub.js

```diff
@@ -203,6 +209,46 @@ class BasicPubSub extends Pubsub {
     this._handleRpcControl(peer, rpc)
   }
 
+  /**
+   * Validates the given message.
+   * @param {Peer} peer
+   * @param {rpc.RPC.Message} message
+   * @returns {Promise<Boolean>}
+   */
+  async validate (peer, message) {
```

Comment:
```suggestion
  async validate (message, peer) {
```

In a semantic perspective, I think that the message should come first as it is what is being validated. Moreover, if we make the message the first parameter and validate if we receive a peer, this will not be a breaking change. We will probably want to add this feature to `floodsub` afterwards, but I think that we should not make this a breaking change, since the topicValidators are also not required.

Reply:
Do you see the topicValidators getting added to `libp2p-pubsub`? I was under the impression that our current `libp2p-pubsub` implementation is not _quite_ up to spec.
https://github.com/libp2p/specs/tree/master/pubsub#topic-validation
Makes it seem like topic-specific validators should be there. I guess its a little ambiguous, you could say that overriding the existing `validate` method is enough to fulfill the spec requirements.

Reply:
Yes, I totally see the topicValidators getting added to `libp2p-pubsub`, but pubsub routers should not be required to used them. At least for now, once we update this in floodsub, we can decide to make it required. That's why I recommended switching the parameters and validate if we have a peer, this way this could be added to libp2p-pubsub but it would not required floodsub to be updated